File: src/main/java/com/johnpier/labproject/models/PostDto.java
package com.johnpier.labproject.models;

import lombok.*;

import java.io.Serializable;
import java.time.LocalDate;

@lombok.Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class PostDto implements Serializable {
    private String id;
    private String label;
    private String description;
    private LocalDate createdAt;
    private ShortUserDto user;
    private String blogId;
    private String content;
}

File: frontend/apps/blog-project/src/app/models/post.dto.ts
import { ShortUserDto } from './user.dto';

export interface PostPreviewDto {
  id: string;
  label: string;
  description: string;
  preview?: string;
  createdAt: string | Date;
  commentsCount: number;
  userId: string;
  blogId: string;
}

export interface PostDto extends PostPreviewDto {
  user: ShortUserDto;
  blogId: string;
  content: string;
}
